examples/workflows/model_nudging.py

```diff
@@ -0,0 +1,105 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# %%
+import os
+from functools import partial
+import torch
+import datetime
+from modulus.distributed.manager import DistributedManager
+from earth2mip.inference_ensemble import run_basic_inference
+from earth2mip.networks import get_model
+from earth2mip.initial_conditions import cds
+
+
+def apply_gaussian_perturbation(
+    x,
+    time_step,
+    in_channel_names,
+    device,
+    latitute_location,
+    latitute_sigma,
+    longitude_location,
+    longitude_sigma,
+    gaussian_amplitude,
+    modified_channels,
+    normalize=False,
+    center=0,
+    scale=1,
+):
+    """Apply a Gaussain nudge of the from
+    A = A₀exp( - (x-x₀)²/(2σ₁²) - (y-y₀)²/(2σ₂²) )
+    with prescribed A₀, x₀, y₀, σ₁, σ₂
+    to user prescribed variable(s)
+    """
+    lat = torch.linspace(-90, 90, x.shape[-2])
+    lon = torch.linspace(-180, 180, x.shape[-1])
+    lat, lon = torch.meshgrid(lat, lon)
+
+    dt = torch.tensor(time_step.total_seconds()) / 86400.0
+
+    gaussian = (
+        dt
+        * gaussian_amplitude
+        * torch.exp(
+            -(
+                (lon - latitute_location) ** 2 / (2 * latitute_sigma**2)
+                + (lat - longitude_location) ** 2 / (2 * longitude_sigma**2)
+            )
+        )
+    )
+
+    if normalize:
+        x += (gaussian.to(device) - center) / scale
+    else:
+        x += gaussian.to(device)
+    return x
```

Comment:
Lots of args...should provide some defaults

Simpler to return perturbation rather than applying it. This allows removing the `time_step` argument.

Doesn't make sense to have 1 amplitude for many channels and the only example below uses just t850.

Don't pass `device`...get it from `x.device` or use functions like `torch.zeros_like`. Pass in lat/lon instead of initializing them (this copying could lead to problems).



```suggestion
def gaussian_perturbation_heating(
    x: torch.Tensor,
    in_channel_names: List[str],
    lat: torch.Tensor,
    lon: torch.Tensor,
    amplitude: float,
    channel_to_perturb: str,
    latitute_location:float =0,
    latitute_sigma:float =50,
    longitude_location: float =0,
    longitude_sigma=10,

):
    """Apply a Gaussain nudge of the from
    A = A₀exp( - (x-x₀)²/(2σ₁²) - (y-y₀)²/(2σ₂²) )
    with prescribed A₀, x₀, y₀, σ₁, σ₂
    to user prescribed variable(s)   
    """
    lat, lon = torch.meshgrid(lat, lon)
    source = torch.zeros_like(x)
    blob = amplitude * torch.exp(
            -(
                (lon - latitute_location) ** 2 / (2 * latitute_sigma**2)
                + (lat - longitude_location) ** 2 / (2 * longitude_sigma**2)
            )
        )
    )
    index = in_channels.index(channel_to_perturb)
    source[:, index] = blob
    return source
```